### ISSUE TITLE
SA1012 is reported for single-line index initializer

### ISSUE THREAD

**Issue by nbarbettini** [DotNetAnalyzers/StyleCopAnalyzers#1768]
SA1012OpeningCurlyBracketsMustBeSpacedCorrectly is reported for the following:

```csharp
public class Class1
{
    public void Test()
    {
        var dictionary = new Dictionary<int, string>() {[0] = "bar" };
    }
}
```

The fix puts a space in between `{ [0]`, like:

```csharp
var dictionary = new Dictionary<int, string>() { [0] = "bar" };
```

This is fine (I think it looks more readable), but unfortunately it conflicts with VS2015's default code formatter. If you use Edit - Format Document (`Ctrl-K, D`) type a semicolon at the end of the line, or paste the code somewhere else, it reverts back to the original.

Not sure which way this should be handled (if at all), but it was annoying.

**Comment by sharwell** [comment]
This sounds like a bug in Roslyn. I recommend filing an issue there, especially if Visual Studio 2013 did not behave this way.

Marking as **external** since the code fix is working properly and the formatter is outside our control. :smile:
